docs/topics/db/aggregation.txt

```diff
@@ -340,29 +340,68 @@ Order of ``annotate()`` and ``filter()`` clauses
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 When developing a complex query that involves both ``annotate()`` and
-``filter()`` clauses, particular attention should be paid to the order
-in which the clauses are applied to the ``QuerySet``.
-
-When an ``annotate()`` clause is applied to a query, the annotation is
-computed over the state of the query up to the point where the annotation
-is requested. The practical implication of this is that ``filter()`` and
-``annotate()`` are not commutative operations -- that is, there is a
-difference between the query::
-
-    >>> Publisher.objects.annotate(num_books=Count('book')).filter(book__rating__gt=3.0)
-
-and the query::
-
-    >>> Publisher.objects.filter(book__rating__gt=3.0).annotate(num_books=Count('book'))
-
-Both queries will return a list of publishers that have at least one good
-book (i.e., a book with a rating exceeding 3.0). However, the annotation in
-the first query will provide the total number of all books published by the
-publisher; the second query will only include good books in the annotated
-count. In the first query, the annotation precedes the filter, so the
-filter has no effect on the annotation. In the second query, the filter
-precedes the annotation, and as a result, the filter constrains the objects
-considered when calculating the annotation.
+``filter()`` clauses, pay particular attention to the order in which the
+clauses are applied to the ``QuerySet``.
+
+When an ``annotate()`` clause is applied to a query, the annotation is computed
+over the state of the query up to the point where the annotation is requested.
+The practical implication of this is that ``filter()`` and ``annotate()`` are
+not commutative operations.
+
+Given:
+
+* Publisher A has two books with ratings 4 and 5
+* Publisher B has two books with ratings 1 and 4
+* Publisher C has one book with rating 1
+
+Here's an example with the ``Count`` aggregate::
+
+    >>> a, b = Publisher.objects.annotate(num_books=Count('book', distinct=True)).filter(book__rating__gt=3.0)
+    >>> a, a.num_books
+    (<Publisher: A>, 2)
+    >>> b, b.num_books
+    (<Publisher: B>, 2)
+
+    >>> a, b = Publisher.objects.filter(book__rating__gt=3.0).annotate(num_books=Count('book'))
+    >>> a, a.num_books
+    (<Publisher: A>, 2)
+    >>> b, b.num_books
+    (<Publisher: B>, 1)
+
+Both queries return a list of publishers that have at least one book with a
+rating exceeding 3.0, hence publisher C is excluded.
+
+In the first query, the annotation precedes the filter, so the filter has no
+effect on the annotation. If we omitted ``distinct=True``, however, the filter
+*would* affect the annotation because it would be calculated on the number of
```

Comment:
Certainly looks the same as 10060. The issue is already documented above in this same file. Given that, I'm inclined to think it's simply a distraction here, and you should just cut everything from "If we omitted..." onwards in this paragraph.

If you want to keep some kind of reference, I would make it very short with a link, e.g. "(Note that `distinct=True` is necessary to avoid a cross-join bug.)" where "cross-join bug" is linked to the "Combining multiple aggregations" section above.